File: starter_kits/Go/src/hlt/Player.go
package hlt

import (
	"errors"
	"fmt"

	"./input"
)

// Player - Structure to hold all information about a player
type Player struct {
	ID       int
	Shipyard *Shipyard
	Halite   int
	Ships    map[int]*Ship
	Dropoffs map[int]*Dropoff
}

func (p *Player) String() string {
	return fmt.Sprintf("Player{ID=%d,Halite=%d}", p.ID, p.Halite)
}

// NewPlayer - Creates a new player from input data
func NewPlayer() *Player {
	var input = input.GetInstance()
	var playerID, _ = input.GetInt()
	var x, _ = input.GetInt()
	var y, _ = input.GetInt()
	return &Player{playerID, NewShipyard(playerID, &Position{x, y}), 0, nil, nil}
}

// GetShip - Get a specific Ship by its ID.
func (p *Player) GetShip(shipId int) (*Ship, error) {
	if ship, ok := p.Ships[shipId]; ok {
		return ship, nil
	}
	return nil, errors.New(fmt.Sprintf("No ship with ID %d", shipId))
}

// Update - Updates the player, reading the ships and dropoffs data
func (p *Player) Update(numShips int, numDropoffs int, halite int) {
	p.Halite = halite
	p.Ships = make(map[int]*Ship)
	p.Dropoffs = make(map[int]*Dropoff)
	for i := 0; i < numShips; i++ {
		var nextShip = NewShip(p.ID)
		p.Ships[nextShip.E.id] = nextShip
	}
	for i := 0; i < numDropoffs; i++ {
		var nextDropoff = NewDropoff(p.ID)
		p.Dropoffs[nextDropoff.E.id] = nextDropoff
	}
}

File: starter_kits/C#-/Halite-3/hlt/Player.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MyBot.hlt
{
    class Player
    {
        public  PlayerId id;
        public  Shipyard shipyard;
        public int halite;
        public Dictionary<EntityId, Ship> ships = new Dictionary<EntityId, Ship>();
        public Dictionary<EntityId, Dropoff> dropoffs = new Dictionary<EntityId, Dropoff>();

        private Player( PlayerId id,  Shipyard shipyard)
        {
            this.id = id;
            this.shipyard = shipyard;
        }

        public void _update( int numShips,  int numDropoffs,  int halite)
        {
            this.halite = halite;

            ships.Clear();
            for (int i = 0; i < numShips; ++i)
            {
                 Ship ship = Ship._generate(id);
                ships.Add(ship.id, ship);
            }

            dropoffs.Clear();
            for (int i = 0; i < numDropoffs; ++i)
            {
                 Dropoff dropoff = Dropoff._generate(id);
                dropoffs.Add(dropoff.id, dropoff);
            }
        }

        public static Player _generate()
        {
             Input input = Input.readInput();

             PlayerId playerId = new PlayerId(input.getInt());
             int shipyard_x = input.getInt();
             int shipyard_y = input.getInt();

            return new Player(playerId, new Shipyard(playerId, new Position(shipyard_x, shipyard_y)));
        }
    }
}
